torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -324,33 +341,11 @@ def partition_parameters(self) -> List[List[Dict]]:
         return self._partition_parameters_cache
 
     @property
-    def _per_device_params(self) -> Dict[torch.device, List[List[Parameter]]]:
+    def _param_to_rank(self) -> Dict[torch.Tensor, int]:
         r"""
-        Sorted list of all the params, first per device then per rank.
-
-        Within a list params are sorted per number of elements to allow for an easy bucketing.
```

Comment:
My understanding was that the sorting here was to ensure a consistent ordering of the parameters per bucket across ranks. As such, now iterating over`self.partition_parameters` in the bucketing should still fulfill this need for a consistent ordering.

Reply:
+1 @blefaudeux please let us know if we were wrong. Thanks!